src/autocat/learning/predictors.py

```diff
@@ -25,173 +24,138 @@ class PredictorError(Exception):
-        if featurization_kwargs is not None:
-            assert isinstance(featurization_kwargs, dict)
-            self._featurization_kwargs = featurization_kwargs.copy()
-            self.featurizer = Featurizer(self.featurizer_class, **featurization_kwargs)
+    @featurizer.setter
+    def featurizer(self, featurizer):
+        if featurizer is not None and isinstance(featurizer, Featurizer):
+            self._featurizer = copy.deepcopy(featurizer)
             if self.is_fit:
                 self.is_fit = False
                 self.X_ = None
                 self.y_ = None
-            self.regressor = self.model_class(
-                **self.model_kwargs if self.model_kwargs else {}
-            )
 
     def copy(self):
         """
         Returns a copy
         """
-        acp = self.__class__(
-            model_class=self.model_class, featurizer_class=self.featurizer_class,
-        )
-        acp.regressor = copy.deepcopy(self.regressor)
+        acp = self.__class__(regressor=self.regressor, featurizer=self.featurizer,)
         acp.is_fit = self.is_fit
-        acp.featurization_kwargs = copy.deepcopy(self.featurization_kwargs)
-        acp.model_kwargs = copy.deepcopy(self.model_kwargs)
 
         return acp
 
+    def to_jsonified_dict(self) -> Dict:
+        featurizer_dict = self.featurizer.to_jsonified_dict()
+        regressor = self.regressor
+        name_string = regressor.__class__.__name__
+        module_string = regressor.__module__
+        try:
+            kwargs = regressor.get_params()
+            _ = json.dumps(kwargs)
+        except TypeError:
+            print("Warning: kwargs not saved")
+            kwargs = None
+        return {
+            "featurizer": featurizer_dict,
+            "regressor": {
+                "name_string": name_string,
+                "module_string": module_string,
+                "kwargs": kwargs,
+            },
+        }
+
+    def write_json_to_disk(self, write_location: str = ".", json_name: str = None):
+        """
+        Writes `Predictor` to disk as a json
+        """
+        jsonified_list = self.to_jsonified_dict()
+
+        if json_name is None:
+            json_name = "predictor.json"
+
+        json_path = os.path.join(write_location, json_name)
+
+        with open(json_path, "w") as f:
+            json.dump(jsonified_list, f)
+
+    @staticmethod
+    def from_jsonified_dict(all_data: Dict):
+        # get regressor
+        name_string = all_data["regressor"].get("name_string")
+        module_string = all_data["regressor"].get("module_string")
+        kwargs = all_data["regressor"].get("kwargs")
+        mod = importlib.import_module(module_string)
+        regressor_class = getattr(mod, name_string)
+        if kwargs is not None:
+            regressor = regressor_class(**kwargs)
+        else:
+            regressor = regressor_class()
+
+        # get featurizer
+        featurizer = Featurizer.from_jsonified_dict(all_data["featurizer"])
```

Comment:
We should have a couple of checks in place, e.g., if `all_data["featurizer"]` is not specified, and raise a suitable error here (unless it gets propagated and will be thrown by the `Featurizer` class; that is fine too but catching errors early is better than later).